cmd/fs_handlers.go

```diff
@@ -158,21 +165,40 @@ func handleStageDirectory(ctx *cli.Context, ctl *client.Client, root, repoRoot s
 		mpb.AppendDecorators(decor.Percentage()),
 	)
 
+	type stageList struct {
+		local    string
+		repoList []string
+	}
+
 	nWorkers := 20
 	start := time.Now()
-	jobs := make(chan stagePair, nWorkers)
+	jobs := make(chan stageList, nWorkers)
 
 	// Start a bunch of workers that will do the actual adding:
 	for idx := 0; idx < nWorkers; idx++ {
 		go func() {
 			for {
-				pair, ok := <-jobs
+				pairSet, ok := <-jobs
 				if !ok {
 					return
 				}
 
-				if err := ctl.Stage(pair.local, pair.repo); err != nil {
-					fmt.Printf("failed to stage %s: %v\n", pair.local, err)
+				firstToStage := ""
+				for i, repoPath := range pairSet.repoList {
+					if i == 0 {
+						firstToStage = repoPath
+						// first occurrence is staged
+						if err := ctl.Stage(pairSet.local, repoPath); err != nil {
+							fmt.Printf("failed to stage %s: %v\n", pairSet.local, err)
+							break
+						}
+						continue
+					}
+					ctl.Copy(firstToStage, repoPath)
```

Comment:
I think you're missing the error handling here.

Reply:
ups, that was left over of copy-pasting. Added a better error handler.